tests/beta/interfaces/test_tape_tf.py

```diff
@@ -0,0 +1,741 @@
+        """Test that the interface is correctly applied multiple times"""
+        with TFInterface.apply(QuantumTape()) as tape:
+            qml.RX(0.5, wires=0)
+            expval(qml.PauliX(0))
+
+        assert tape.interface == "tf"
+        assert isinstance(tape, TFInterface)
+        assert tape.__bare__ == QuantumTape
+        assert tape.dtype is tf.float64
+
+        TFInterface.apply(QuantumTape(), dtype=tf.float32)
+        assert tape.interface == "tf"
+        assert isinstance(tape, TFInterface)
+        assert tape.__bare__ == QuantumTape
+        assert tape.dtype is tf.float64
+
+    def test_get_parameters(self):
+        """Test that the get parameters function correctly sets and returns the
+        trainable parameters"""
+        a = tf.Variable(0.1)
+        b = tf.constant(0.2)
+        c = tf.Variable(0.3)
+        d = 0.4
+
+        with tf.GradientTape() as tape:
+            with TFInterface.apply(QuantumTape()) as qtape:
+                qml.Rot(a, b, c, wires=0)
+                qml.RX(d, wires=1)
+                qml.CNOT(wires=[0, 1])
+                expval(qml.PauliX(0))
+
+        assert qtape.trainable_params == {0, 2}
+        assert np.all(qtape.get_parameters() == [a, c])
+
+    def test_execution(self):
+        """Test execution"""
+        a = tf.Variable(0.1)
+        dev = qml.device("default.qubit", wires=1)
+
+        with tf.GradientTape() as tape:
+            with TFInterface.apply(QuantumTape()) as qtape:
+                qml.RY(a, wires=0)
+                qml.RX(0.2, wires=0)
+                expval(qml.PauliZ(0))
+
+            assert qtape.trainable_params == {0}
+            res = qtape.execute(dev)
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (1,)
+
+    def test_jacobian(self, mocker, tol):
+        """Test jacobian calculation"""
+        spy = mocker.spy(QuantumTape, "jacobian")
+        a = tf.Variable(0.1, dtype=tf.float64)
+        b = tf.Variable(0.2, dtype=tf.float64)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        with tf.GradientTape() as tape:
+            with TFInterface.apply(QuantumTape()) as qtape:
+                qml.RY(a, wires=0)
+                qml.RX(b, wires=1)
+                qml.CNOT(wires=[0, 1])
+                expval(qml.PauliZ(0))
+                expval(qml.PauliY(1))
+
+            assert qtape.trainable_params == {0, 1}
+            res = qtape.execute(dev)
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2,)
+
+        expected = [tf.cos(a), -tf.cos(a) * tf.sin(b)]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = tape.jacobian(res, [a, b])
+        expected = [[-tf.sin(a), tf.sin(a) * tf.sin(b)], [0, -tf.cos(a) * tf.cos(b)]]
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        spy.assert_called()
+
+    def test_jacobian_dtype(self, tol):
+        """Test calculating the jacobian with a different datatype"""

```

Comment:
What would this datatype be different from? The previous test's `tf.float64`? Might be worth clarifying

Reply:
```suggestion
        """Test calculating the jacobian with a different datatype. Here, we
        specify tf.float32, as opposed to the default value of tf.float64."""
```